ssr-express/README.md

```diff
@@ -0,0 +1,131 @@
+<!---
+Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+## Introduction
+
+amp-ssr-express is an [express](http://expressjs.com/) middleware that applies AMP server-side-rendering on the fly, using the [amp-ssr](../ssr) component.
+
+For the middleware to work, a route that renders AMP pages must be available. It will then rewrite the
+request to that route and apply server-side-rendering to the result.
+
+As an example, a website that has an SSR page at `https://example.com/page.html` would have it's AMP
```

Comment:
I don't quite follow this paragraph. What does it mean for `/page.html` to be
rewritten to `/amp/page.html`?

What I was expecting was something like the below.

Before:

| URL | |
| --- | --- |
| `/page.html` | Canonical AMP |
| `/amp/page.html` | Does not exist |

After:

| URL | |
| --- | --- |
| `/page.html` | SSR'd AMP as transformed by `amp-ssr` (not valid AMP), contains `<link rel=amphtml href=/amp/page.html>` |
| `/amp/page.html` | Valid AMP, almost exactly the same as "before" version of `/page.html`, except it contains `<link rel=canonical href=/page.html>` and relative paths are rewritten. |

Reply:
Hum. This is not how it is working at the moment, but I think this is a better approach than the current one. @sebastianbenz WDYT?

Reply:
The adding/removing directories to fix URLs also seems a bit risky--even in AMP I'm not sure that you will be able to correctly rewrite all e.g. `../img/foo.png` to `../amp/img/foo.png` or whatever. (AMP version maybe better as `page.amp.html`, so that relative URLs don't need to change.)

Reply:
Good point Michael, we definitely should avoid re-writing the path!